lib/test_cleanup.rb

```diff
@@ -104,11 +108,15 @@ def cleanup_shared
 
     private
 
-    def checkout_branch_if_needed(repository, branch = "master")
+    def checkout_branch_if_needed(repository)
+      default_branch = Utils.safe_popen_read(
+        git, "-C", repository, "symbolic-ref", "refs/remotes/origin/HEAD", "--short"
+      ).strip.split("/", 2).last
       current_branch = Utils.safe_popen_read(
-        git, "-C", repository, "symbolic-ref", "HEAD"
+        git, "-C", repository, "symbolic-ref", "HEAD", "--short"
```

Comment:
Why the `--short` addition?

Reply:
Without the short this is 

```
$ git symbolic-ref HEAD
refs/heads/master
```

Whereas what we're really looking for is just "master" I can't see how it could work currently, but perhaps that's just something about my machine.